### PR TITLE
Add `shape` option in examples

### PR DESCRIPTION
**Pull request by georgebisbas**
Adding shape option in examples.
Overwrites previous `-nd` option as a length of the tuple.
In general adds another enhanced option.

### CODE REVIEW

File: examples/seismic/acoustic/acoustic_example.py [hunk 1]
```diff
@@ -66,8 +66,8 @@ def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
 if __name__ == "__main__":
     description = ("Example script for a set of acoustic operators.")
     parser = ArgumentParser(description=description)
-    parser.add_argument('-nd', dest='ndim', default=3, type=int,
-                        help="Preset to determine the number of dimensions")
+    parser.add_argument("-d", "--shape", default=(51, 51, 51), type=int, nargs="+",
+                        help="Determine the grid size")
```
Comment (FabioLuporini) [inline examples/seismic/acoustic/acoustic_example.py:70]: "Determine the grid size" -> "The grid shape"

File: examples/seismic/acoustic/acoustic_example.py [hunk 2]
```diff
@@ -90,9 +90,10 @@ def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
     args = parser.parse_args()
 
     # 3D preset parameters
-    shape = tuple(args.ndim * [51])
-    spacing = tuple(args.ndim * [15.0])
-    tn = 750. if args.ndim < 3 else 250.
+    shape = args.shape
+    ndim = len(shape)
+    spacing = tuple(ndim * [15.0])
+    tn = 750. if ndim < 3 else 250.
```
Comment (FabioLuporini) [inline examples/seismic/acoustic/acoustic_example.py:96]: blank line missing

File: examples/seismic/elastic/elastic_example.py [hunk 3]
```diff
@@ -61,16 +61,11 @@ def test_elastic():
...
-    if args.dim2:
-        shape = (150, 150)
-        spacing = (10.0, 10.0)
-        tn = 750.0
-    # 3D preset parameters
-    else:
-        shape = (150, 150, 150)
-        spacing = (10.0, 10.0, 10.0)
-        tn = 1250.0
+    # preset parameters
```
Comment (FabioLuporini) [inline examples/seismic/elastic/elastic_example.py:64]: all comments should start with capital letter

File: examples/seismic/acoustic/acoustic_example.py (hunk 1 above)
Comment (mloubout) [reply examples/seismic/acoustic/acoustic_example.py:70]: This should be an extra argument with nd overwritten if shape is provided not replacing it.
`-d` is not a very good name IMO, would enforce `--shape`

Comment (FabioLuporini) [reply examples/seismic/acoustic/acoustic_example.py:70]: I think the `-d` is for legacy reasons -- `benchmark.py` takes `-d` for the shape, AFAIR. I don't mind keeping it or removing it. Your call guys

Comment (georgebisbas) [reply examples/seismic/acoustic/acoustic_example.py:70]: I did inherit from benchmark. @mloubout I will add it as extra , OK.

File: examples/seismic/acoustic/acoustic_example.py (hunk 2 above)
Comment (georgebisbas) [reply examples/seismic/acoustic/acoustic_example.py:96]: Fixed

File: examples/seismic/elastic/elastic_example.py (hunk 3 above)
Comment (georgebisbas) [reply examples/seismic/elastic/elastic_example.py:64]: Fixed

File: examples/seismic/acoustic/acoustic_example.py (hunk 1 above)
Comment (georgebisbas) [reply examples/seismic/acoustic/acoustic_example.py:70]: Added as extra argument.

File: examples/seismic/viscoelastic/viscoelastic_example.py [hunk 4]
```diff
@@ -45,8 +45,10 @@ def test_viscoelastic():
 if __name__ == "__main__":
     description = ("Example script for a set of viscoelastic operators.")
     parser = ArgumentParser(description=description)
-    parser.add_argument('--2d', dest='dim2', default=False, action='store_true',
-                        help="Preset to determine the physical problem setup")
+    parser.add_argument("-nd", dest='ndim', default=3, type=int,
+                        help="Preset to determine the number of dimensions")
+    parser.add_argument("-d", "--shape", default=(150, 150, 150), type=int, nargs="+",
+                        help="Determine the grid shape, defaults in (153, 153, 153)")
```
Comment (FabioLuporini) [inline examples/seismic/viscoelastic/viscoelastic_example.py:51]: this is not consistent, you write the defaults here but not in the other examples. I think you can safely drop it, the --help should display it already

File: examples/seismic/acoustic/acoustic_example.py [hunk 5]
```diff
@@ -66,8 +66,10 @@ def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
 if __name__ == "__main__":
     description = ("Example script for a set of acoustic operators.")
     parser = ArgumentParser(description=description)
-    parser.add_argument('-nd', dest='ndim', default=3, type=int,
+    parser.add_argument("-nd", dest="ndim", default=3, type=int,
                         help="Preset to determine the number of dimensions")
+    parser.add_argument("-d", "--shape", default=(51, 51, 51), type=int, nargs="+",
+                        help="Determine the grid shape")
```
Comment (FabioLuporini) [inline examples/seismic/acoustic/acoustic_example.py:72]: "Determine" here sounds a bit off to me

Comment (georgebisbas) [reply examples/seismic/acoustic/acoustic_example.py:72]: Set, Define, I chose determine to be consistent with the already existing help in line 70.

File: examples/seismic/viscoelastic/viscoelastic_example.py (hunk 4 above)
Comment (georgebisbas) [reply examples/seismic/viscoelastic/viscoelastic_example.py]: Fixed

**Pull request merged**